source/agora/consensus/protocol/Nominator.d

```diff
@@ -706,15 +706,11 @@ extern(D):
         const block = this.ledger.getBlocksFrom(Height(block_sig.height)).front;
         if (!this.collectBlockSignature(block_sig, block.hashFull()))
             return BlockHeader.init;
-        const signed_block = this.updateMultiSignature(block);
-        if (signed_block == Block.init)
-        {
-            log.trace("Failed to add signature {} for block {} utxo {}",
-                block_sig.signature, block_sig.height, block_sig.utxo);
-            return BlockHeader.init;
-        }
-        this.ledger.updateBlockMultiSig(signed_block.header);
-        return signed_block.header;
+        const updated_sig = this.updateMultiSignature(block.header);
+        if (updated_sig == BlockHeader.init) // No signatures were added
+            return block.header;
```

Comment:
Can you explain why this line is changed from `BlockHeader.init`?

Reply:
If we call updateMultiSignature but have no stored signatures we return `BlockHeader.init` as seen here:
https://github.com/bosagora/agora/blob/5cd9f7a8e71b15ee010f347a79f57c6fb1b23dbe/source/agora/consensus/protocol/Nominator.d#L1049-L1053
But we want to still return the header we received with the signatures it had included.

Reply:
Now that I check there is another call to `updateMultiSignature` that I forgot to check for BlockHeader.init returned.
I will add a fix.

Reply:
Which call?

Reply:
https://github.com/hewison-chris/agora/blob/d614d1f4c6e6ec96948a7ff031cc9466b4b2eff9/source/agora/consensus/protocol/Nominator.d#L997

Reply:
In fact I think I should return the original header from `updateMultiSignature`. I just had some problems with casting but will try again.

Reply:
I pushed an update which clones the `const(BlockHeader)` to `BlockHeader` in `updateMultiSignature` to be returned if no signatures are stored.

Reply:
Another update to return `const(BlockHeader)` from `updateMultiSignature ` and move the clone to mutable to `updateHeader` function of `Block`. I think this is better.